source/s3/s3.go

```diff
@@ -0,0 +1,61 @@
+package s3
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/MiLk/nsscache-go/cache"
+	"github.com/MiLk/nsscache-go/source"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+	"github.com/pkg/errors"
+)
+
+type S3Source struct {
+	prefix string
+	bucket string
+	client s3iface.S3API
+}
+
+func CreateS3Source(client s3iface.S3API, prefix string, bucket string) source.Source {
+	return &S3Source{
+		client: client,
+		prefix: prefix,
+		bucket: bucket,
+	}
+}
+
+func (s *S3Source) run(key string, e cache.Entry, c *cache.Cache) error {
```

Comment:
How does this function handle several entries per file?

Reply:
It doesn't :(